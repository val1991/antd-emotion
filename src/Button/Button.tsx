/** @jsxImportSource @emotion/react */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { jsx } from '@emotion/react'
import { ReactElement } from 'react'
import { Button as AntButton, ButtonProps } from 'antd'

export function Button({
  children,
  ...rest
}: ButtonProps): ReactElement {


  return (
    <AntButton css={{
      '&.ant-btn': {
        background: 'red',

         '&:active': {
          background: 'green'
        },

        '&:hover': {
          background: 'aqua'
        },

        '&:disabled': {
          background: 'grey'
        },
      }
    }}  {...rest}>
      {children}
    </AntButton>
  )
}
