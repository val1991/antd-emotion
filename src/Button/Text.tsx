/** @jsxImportSource @emotion/react */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { jsx } from '@emotion/react'
import { Col } from 'antd'
import { PropsWithChildren, ReactElement } from 'react'

export function Text({
  children,
}: PropsWithChildren): ReactElement {


  return (
    <Col
      css={{
        color: 'red',

        '&:active': {
          background: 'green'
        },

        '&:hover': {
          background: 'aqua'
        },
      }}
    >
      {children}
    </Col>
  )
}
