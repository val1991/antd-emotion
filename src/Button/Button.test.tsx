import { render, screen } from '@testing-library/react';
import { Button } from './Button';
import { createSerializer } from '@emotion/jest';
import { StyleProvider } from '@ant-design/cssinjs';

expect.addSnapshotSerializer(createSerializer())

const onClick = jest.fn();

test('Button styles', () => {
  const styleHolder = document.createElement('div');
  const { asFragment } = render(
    <StyleProvider container={styleHolder}>
  <Button onClick={onClick}>test</Button>
   </StyleProvider>
  );
  expect(asFragment()).toMatchSnapshot()
  
  const button = screen.getByRole('button');
  expect(button).toBeInTheDocument();

   expect(button).toHaveStyle({
      background: 'red'
    })
});
