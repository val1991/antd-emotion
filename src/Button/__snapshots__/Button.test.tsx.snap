// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button styles 1`] = `
<DocumentFragment>
  .emotion-1.ant-btn {
  background: red;
}

.emotion-1.ant-btn:active {
  background: green;
}

.emotion-1.ant-btn:hover {
  background: aqua;
}

.emotion-1.ant-btn:disabled {
  background: grey;
}

<button
    class="ant-btn emotion-0 ant-btn-default emotion-1"
    type="button"
  >
    <span>
      test
    </span>
  </button>
</DocumentFragment>
`;
